velox/common/compression/Lz4Compression.h

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <lz4.h>
+#include <lz4frame.h>
+#include <lz4hc.h>
+#include <memory>
+#include "velox/common/compression/Compression.h"
+#include "velox/common/compression/HadoopCompressionFormat.h"
+
+namespace facebook::velox::common {
+
+struct Lz4CodecOptions : CodecOptions {
+  enum Type { kLz4Frame, kLz4Raw, kLz4Hadoop };
+
+  Lz4CodecOptions(
+      Type type,
+      int32_t compressionLevel = kUseDefaultCompressionLevel)
+      : CodecOptions(compressionLevel), type(type) {}
+
+  Type type;
+};
+
+class Lz4CodecBase : public Codec {
+ public:
+  explicit Lz4CodecBase(int32_t compressionLevel);
+
+  int32_t minimumCompressionLevel() const override;
+
+  int32_t maximumCompressionLevel() const override;
+
+  int32_t defaultCompressionLevel() const override;
+
+  int32_t compressionLevel() const override;
+
+  CompressionKind compressionKind() const override;
+
+ protected:
+  const int32_t compressionLevel_;
+};
+
+class Lz4FrameCodec : public Lz4CodecBase {
```

Comment:
do we need to expose all these classes in the header, or could we have them only in the .cpp file and expose a factory function?